bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/internal/findandreplace/FindReplaceLogic.java

```diff
@@ -597,7 +598,10 @@ public boolean performSelectAndReplace(String findString, String replaceString)
 		if (!isFindStringSelected(findString)) {
 			performSearch(findString);
 		}
-		return replaceSelection(replaceString);
+		if (getStatus() instanceof NoStatus) {
```

Comment:
getStatus().wasSuccessful()

Reply:
You have removed this added check with one of the previous commits. Was that intended?